pkg/steps/cluster_claim.go

```diff
@@ -93,18 +94,28 @@ func (s *clusterClaimStep) run(ctx context.Context) error {
 	}
 
 	clusterClaim, err := s.acquireCluster(ctx, waitForClaim)
+	// Make sure we release the claim no matter what. This is a very brute force solution,
+	// that works even if wrapped() blocks and doesn't correctly end when ctx is cancelled.
+	if clusterClaim != nil {
+		go func() {
+			<-ctx.Done()
+			if err := s.releaseCluster(cleanupCtx, clusterClaim, false); err != nil {
+				logrus.WithError(err).Error("failed to release cluster claim")
+			}
+		}()
+	}
 	if err != nil {
 		acquireErr := results.ForReason("acquiring_cluster_claim").ForError(err)
 		// always attempt to delete claim if one exists
 		var releaseErr error
 		if clusterClaim != nil {
-			releaseErr = results.ForReason("releasing_cluster_claim").ForError(s.releaseCluster(ctx, clusterClaim, true))
+			releaseErr = results.ForReason("releasing_cluster_claim").ForError(s.releaseCluster(cleanupCtx, clusterClaim, true))
```

Comment:
TIL: Nice.
I was looking for such a thing the last time you told me the issue.
Also wondering if we should the same for Boskos accquire: what happened to an acquired resource from Boskos if the test is aborted? I guess the current implementation also does the clean up based on some timeout.



https://github.com/openshift/ci-tools/blob/83e16a4b6b391f438d57de976689624617a7d418/pkg/steps/pod.go#L33-L36